extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriGatewayHandler.java

```diff
@@ -80,16 +81,48 @@ public void handleCommand(ChannelUID channelUID, Command command) {
 
     @Override
     public void initialize() {
+        scheduler.execute(this::initializeAsync);
+    }
+    
+    private synchronized void initializeAsync() {
         TradfriGatewayConfig configuration = getConfigAs(TradfriGatewayConfig.class);
-        if (configuration.host == null || configuration.host.isEmpty()) {
+
+        if (isNullOrEmpty(configuration.host)) {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
                     "Host must be specified in the configuration!");
             return;
         }
-        if (configuration.code == null || configuration.code.isEmpty()) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                    "Security code must be provided in the configuration!");
-            return;
+        if (isNullOrEmpty(configuration.code)) {
+            if (isNullOrEmpty(configuration.identity) || isNullOrEmpty(configuration.preSharedKey)) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Either security code or identity and pre-shared key must be provided in the configuration!");
+                return;
+            }
+        } else {
+            configuration.identity = UUID.randomUUID().toString().replace("-", "");
+            configuration.preSharedKey = authenticateOnGateway(configuration);
+
+            if (isNullOrEmpty(configuration.preSharedKey)) {
+                if (isOldFirmware()) {
+                    /*
+                     * older firmware - fall back to authentication with security code
+                     * in this case the Thing configuration will not be persisted
+                     */
+                    logger.warn("Gateway with old firmware - please consider upgrading to the latest version.");
+                    configuration.preSharedKey = configuration.code;
+                    configuration.identity = "";
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Pre-shared key was not obtain sucessfully");
+                    return;
+                }
+            } else {
+                logger.info(
```

Comment:
My point about the info logging was that the user can take identity+psk and paste it in his config. Having the logging without the PSK therefore does not help.

Reply:
My thought was that the key is 'sensitive' information and it should not be shown in the log, and the experienced enough users will be able to find it in the thing configuration in Paper UI - but if you say - I'm totally fine to also log it - what severity should I use - debug (or info would also be OK) ?

Reply:
I understand your point and I am a bit undecided what we should do. Note that the experienced user that uses textual configuration might not even have the Paper UI installed. In other bindings, the authentication is e.g. handled through a console command. So 
But I agree that a "standard" log file must not contain any passwords; so I agree that INFO logging is not the best here. But asking people to change their logging to DEBUG in order to get hold of the PSK is pretty awkward and rather useless.
Maybe the best would be to give the users an easy way to see the config of a Thing through a console command - but this is clearly out of scope of this PR.

So eventually my suggestion would be to do a DEBUG logging of identity+psk, I hope your security concerns are addressed through this.